### ISSUE TITLE
--aot flag throws error Unexpected value 'TreeModule' imported by the module 'AnnotateModule'

### ISSUE THREAD

**Issue by areijngoudt** [angular/angular-cli#2957]
cli 1.0.0-beta.18 on OSX Sierra.

Using ng build --prod --aot throws this error:

Unexpected value 'TreeModule' imported by the module 'AnnotateModule'

AnnotateModule:
```
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TreeModule } from 'angular2-tree-component';

import { SharedModule } from '../shared/shared.module';

import { routing } from './annotate.routing';
import { AnnotateComponent } from './annotate.component';

@NgModule({
  imports: [
    CommonModule,
    TreeModule,
    SharedModule,
    routing,
  ],
  exports: [
  ],
  declarations: [
    AnnotateComponent,
  ],
})
export class AnnotateModule {
}
```

The Treeview is a thrid party component https://github.com/500tech/angular2-tree-component

**Comment by marisveide** [comment]
For me too, similarily:
$ ng build --prod --aot
```
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
  0% compilingUnexpected value 'LinkyModule' imported by the module 'MyModule'
Error: Unexpected value 'LinkyModule' imported by the module 'MyModule'
    at .../node_modules/@angular/compiler/bundles/compiler.umd.js:14326:37
    at Array.forEach (native)
    at CompileMetadataResolver.getNgModuleMetadata (.../node_modules/@angular/compiler/bundles/compiler.umd.js:14311:46)
    at .../node_modules/@angular/compiler/bundles/compiler.umd.js:13284:43
    at Array.forEach (native)
    at _extractModulesAndPipesOrDirectives (.../node_modules/@angular/compiler/bundles/compiler.umd.js:13283:28)
    at analyzeNgModules (.../node_modules/@angular/compiler/bundles/compiler.umd.js:13039:16)
    at OfflineCompiler.compileModules (.../node_modules/@angular/compiler/bundles/compiler.umd.js:13115:20)
    at CodeGenerator.codegen (/Users/iminar/Dev/angular/modules/@angular/compiler-cli/src/codegen.ts:71:26)
    at AotPlugin._make (.../node_modules/@ngtools/webpack/src/plugin.js:186:43)
    at Compiler.<anonymous> (.../node_modules/@ngtools/webpack/src/plugin.js:151:75)
    at Compiler.applyPluginsParallel (.../node_modules/tapable/lib/Tapable.js:156:14)
    at Compiler.<anonymous> (.../node_modules/webpack/lib/Compiler.js:463:8)
    at Compiler.applyPluginsAsync (.../node_modules/tapable/lib/Tapable.js:73:70)
    at Compiler.compile (.../node_modules/webpack/lib/Compiler.js:456:7)
    at Compiler.<anonymous> (.../node_modules/webpack/lib/Compiler.js:219:10)
    at Compiler.readRecords (.../node_modules/webpack/lib/Compiler.js:367:10)
    at Compiler.<anonymous> (.../node_modules/webpack/lib/Compiler.js:216:9)
    at next (.../node_modules/tapable/lib/Tapable.js:81:11)
    at Compiler.<anonymous> (.../node_modules/webpack/lib/CachePlugin.js:34:58)
    at Compiler.applyPluginsAsync (.../node_modules/tapable/lib/Tapable.js:85:13)
    at Compiler.<anonymous> (.../node_modules/webpack/lib/Compiler.js:213:8)
    at next (.../node_modules/tapable/lib/Tapable.js:81:11)
    at Compiler.<anonymous> (.../node_modules/webpack/lib/node/NodeEnvironmentPlugin.js:23:3)
    at Compiler.applyPluginsAsync (.../node_modules/tapable/lib/Tapable.js:85:13)
    at Compiler.run (.../node_modules/webpack/lib/Compiler.js:210:7)
    at .../node_modules/angular-cli/tasks/build-webpack.js:26:29
    at Class.run (.../node_modules/angular-cli/tasks/build-webpack.js:25:16)
    at Class.run (.../node_modules/angular-cli/commands/build.js:50:26)
    at Class.<anonymous> (.../node_modules/angular-cli/lib/models/command.js:152:17)
    at tryCatch (.../node_modules/rsvp/dist/lib/rsvp/-internal.js:215:12)
    at invokeCallback (.../node_modules/rsvp/dist/lib/rsvp/-internal.js:230:13)
    at publish (.../node_modules/rsvp/dist/lib/rsvp/-internal.js:198:7)
    at flush (.../node_modules/rsvp/dist/lib/rsvp/asap.js:85:5)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

When building with ng build --prod it all builds just fine.

**Comment by Meligy** [comment]
This is not an Angular CLI issue.

Here's the thing. Libraries need to do extra stuff to be AoT friendly in general. Unfortunately angular2-tree-component is not AoT ready yet.

There's an open PR https://github.com/500tech/angular2-tree-component/pull/62 that's not merged yet. There are some other efforts with appearantly conflicting outcomes shown in https://github.com/500tech/angular2-tree-component/issues/68 for angular2-tree-component 1.3.4 and 1.3.5.

**Comment by areijngoudt** [comment]
This problem is solved in the angular2-tree-component 1.3.5
